### PR TITLE
[cli] add Doxygen tags to all `trel` commands

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -7778,18 +7778,97 @@ template <> otError Interpreter::Process<Cmd("mac")>(Arg aArgs[])
...
         error = ProcessEnableDisable(aArgs + 1, otTrelIsFilterEnabled, otTrelSetFilterEnabled);
     }
+    /**
+     * @cli trel peers
+     * @code
+     * trel peers
+     * | No  | Ext MAC Address  | Ext PAN Id       | IPv6 Socket Address                              |
+       +-----+------------------+------------------+--------------------------------------------------+
+       |   1 | 5e5785ba3a63adb9 | f0d9c001f00d2e43 | [fe80:0:0:0:cc79:2a29:d311:1aea]:9202            |
+       |   2 | ce792a29d3111aea | dead00beef00cafe | [fe80:0:0:0:5c57:85ba:3a63:adb9]:9203            |
```
Comment (abtink) [inline src/cli/cli.cpp:7857]: Add missing `*`?

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -7778,18 +7778,97 @@ template <> otError Interpreter::Process<Cmd("mac")>(Arg aArgs[])
...
+     * Done
+     * @endcode
+     * @code
+     * trel peers list
+     * 001 ExtAddr:5e5785ba3a63adb9 ExtPanId:f0d9c001f00d2e43 SockAddr:[fe80:0:0:0:cc79:2a29:d311:1aea]:9202
+       002 ExtAddr:ce792a29d3111aea ExtPanId:dead00beef00cafe SockAddr:[fe80:0:0:0:5c57:85ba:3a63:adb9]:9203
+       >>>>>>> [trel] implement new TREL model using DNS-SD
```
Comment (abtink) [inline src/cli/cli.cpp:7864]: Remove this line?
```suggestion
   
```

**Review by abtink — COMMENTED** [review]
Looks good. Two small suggestions below. Thanks.

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -7778,18 +7778,97 @@ template <> otError Interpreter::Process<Cmd("mac")>(Arg aArgs[])
...
+ * @cli trel
+ * @code
+ * trel
+ * Enabled
+ * Done
+ * @endcode
+ * @par
+ * Indicates whether the TREL radio operation is enabled.
+ * @note `OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE` is required for all `trel` sub-commands.
+ * @sa otTrelIsEnabled
```
Comment (Vyrastas) [inline src/cli/cli.cpp:7791]: We can do an `@par api_copy` for this one and still keep the `@note`

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -7778,18 +7778,95 @@ template <> otError Interpreter::Process<Cmd("mac")>(Arg aArgs[])
...
+     * @par
+     * Indicates whether TREL filter mode is enabled.
+     * @par
+     * When filter mode is enabled, all Rx and Tx traffic sent through the TREL interface gets silently dropped.
+     * @note This mode is used mostly for testing.
+     * @sa otTrelIsFilterEnabled
+     */
+
+    /**
+     * @cli trel filter (enable,disable)
```
Comment (Vyrastas) [inline src/cli/cli.cpp:7831]: Can we move this block down so it's just before the ProcessEnableDisable line?  (similar to how you have the `trel` / `trel (enable,disable)` ones above)

**Review by Vyrastas — APPROVED** [review]

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

**Review by jwhui — APPROVED** [review]

**Pull request merged**